apps/api/src/app/shared/interceptors/invalidate-cache.interceptor.ts

```diff
@@ -2,13 +2,15 @@ import { getInvalidateQuery } from './shared-cache';
 import { CacheKeyPrefixEnum, CacheService, invalidateCache } from '../services/cache';
 import { Inject } from '@nestjs/common';
 
+const USE_CASE_METHOD = 'execute';
+
 // eslint-disable-next-line @typescript-eslint/naming-convention
 export function InvalidateCache(storeKeyPrefix?: CacheKeyPrefixEnum | CacheKeyPrefixEnum[]) {
   const injectCache = Inject(CacheService);
 
   return (target: any, key: string, descriptor: any) => {
     const originalMethod = descriptor.value;
-    const methodName = key;
+    const methodName = key === USE_CASE_METHOD ? target.constructor.name : key;
```

Comment:
I think this can be removed

Reply:
It is still needed in order to check if the query credentials will come from the create query response or a command object.